### PR TITLE
Add Test to enumerate dynamically generated assemblies

### PR DESCRIPTION
**Pull request by gkhanna79**
Test for validating scenario in https://github.com/dotnet/corefx/issues/18877. Should be merged after https://github.com/dotnet/coreclr/pull/11294 comes into CoreFX.

@jkotas PTAL.

### CODE REVIEW

File: src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs [hunk 1]
```diff
@@ -123,6 +123,20 @@ public static void LoadRefEmitAssemblyInLoadContext(AssemblyLoadContext loadCont
 
             // Load context of the statically loaded assembly is the custom load context in which dynamic assembly was created
             Assert.Equal(loadContextRefEmitAssembly, AssemblyLoadContext.GetLoadContext((Assembly)asmRefEmitLoadedStatic));
+
+            // Enumerate the assemblies in the AppDomain and confirm that the Dynamically generated assembly is present.
+            var allAssemblies = AppDomain.CurrentDomain.GetAssemblies();
```
Comment (stephentoub) [inline src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs]: Nit: var => Assembly[]

File: src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs [hunk 2]
```diff
@@ -123,6 +123,20 @@ public static void LoadRefEmitAssemblyInLoadContext(AssemblyLoadContext loadCont
...
+            bool fDynamicAssemblyFound = false;
```
Comment (danmosemsft) [inline src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs]: style says avoid hungarian prefixes in corefx...

File: src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs [hunk 3]
```diff
@@ -123,6 +123,20 @@ public static void LoadRefEmitAssemblyInLoadContext(AssemblyLoadContext loadCont
...
+            foreach (Assembly asm in allAssemblies)
+            {
+                if (asm.FullName == asmRefEmitLoaded.FullName)
```
Comment (jkotas) [inline src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs]: Do we actually want to verify that it is exactly same assembly, not just that the full name matches?

**Comment by jkotas** [comment]
LGTM modulo comments.

File: src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs (hunk 3 above)
Comment (gkhanna79) [inline src/System.Runtime.Loader/tests/RefEmitLoadContext/RefEmitLoadContextTest.cs]: Pushed change to check for assembly reference as well.

**Pull request merged**